sds_data_manager/lambda_code/SDSCode/upload_api.py

```diff
@@ -50,25 +50,30 @@ def lambda_handler(event, context):
 
     :return: A pre-signed url where users can upload a data file to the SDS.
     """
-    logger.info(f"Event: {event}")
-    logger.info(f"Context: {context}")
+    path_params = event.get("pathParameters", {}).get("proxy", None)
+    logger.debug("Parsing path parameters=[%s] from event=" "[%s]", path_params, event)
 
-    if "filename" not in event["queryStringParameters"]:
+    if not path_params:
         return {
             "statusCode": 400,
-            "body": json.dumps("Please specify a filename to upload"),
+            "body": json.dumps(
+                "No filename given for upload. "
+                "Please provide a filename "
+                "in the path. Eg. /upload/path/to/file/filename.pkts"
+            ),
         }
-
-    filename = event["queryStringParameters"]["filename"]
-
-    science_file = ScienceFilepathManager(filename)
-
-    if not science_file.is_valid:
-        return {"statusCode": 400, "body": science_file.error_message}
+    # TODO: Handle other filetypes other than science files
+    #      The current ScienceFilepathManager only accepts filenames, not the full path
+    filename = os.path.basename(path_params)
```

Comment:
Do we anticipate adding support for filepaths before our demo next week? Just wondering so I can make sure the API documentation matches what we support. Currently the API docs specify that filepaths are valid.

Reply:
This is somewhat confusing to read at first. But the `SciencFilepathManager` only accepts the filenames, our API accepts the full path. This `basename` is here to strip off the front path part for now and why I put the TODO in. So, to summarize we currently accept full paths (or just filenames too), but only full paths to science file naming conventions if that makes sense.

Reply:
Yes, that makes sense. Thanks for clarifying!